src/components/general/list/index.tsx

```diff
@@ -21,14 +21,21 @@ type ProductItem = {
   id: string;
 };
 
-type Items = ProductItem[];
+type Items = ProductItem[] | React.ReactElement[];
```

Comment:
I think we're creating a bit of coupling here. This was initially created to display search items and then we realised that it can be used more generally to display list of things, which is a good thing. The way to do it with the least possible coupling is to "tell" each inner element/component to render itself. So it might be a good idea to refactor the search list so they all confirm to the same contract and the list doesn't need to care about what it's rendering. Not sure if it needs to be done on this PR though. What do you think of this as an approach?

Reply:
Yes that approach makes more sense, its now not really a general list component!

Reply:
Thinking about making the List component more general, I've removed the data-testid ternary from the cart list test. 

I think you're right in refactoring search List and the List component could be another ticket, shall I create one?

Reply:
Done!